docs/libcudacxx/ptx/examples/device_tensormap_initialization.rst

```diff
@@ -0,0 +1,141 @@
+
+In this kernel, the tensor map template is passed as a kernel parameter. This is
+the preferred way of moving the template from the host to device. If the kernel
+is intended to update an existing tensor map in device memory, it can take a
+pointer to the existing tensor map to modify.
+
+**Note**: The format of the tensor map may change from one GPU architecture to
+the next. Therefore, the ``tensormap.replace`` instructions are marked as specific
+to ``sm_90a``. To use them, compile using ``nvcc -arch sm_90a ...``.
+
+.. literalinclude:: ./device_tensormap_initialization.cu
+    :language: c++
+    :dedent:
+    :start-after: example-begin modification
+    :end-before: example-end modification
+
+Usage of a modified tensor map
+------------------------------
+
+In contrast to using a tensor map that is passed as a ``const __grid_constant__``
+kernel parameter, using a tensor map in global memory requires establishing a
+release-acquire pattern in the tensor map proxy between the threads that modify
+the tensor map and the threads that use it.
+
+The release part of the pattern was shown in the previous section. It is
+accomplished using the :ref:`tensormap.cp_fenceproxy
+<libcudacxx-ptx-instructions-tensormap-cp_fenceproxy>` instruction.
+
+The acquire part is accomplished using the :ref:`fence.proxy.tensormap::generic
+<libcudacxx-ptx-instructions-fence>` instruction. If the two threads are on the
+same device, the scope ``.gpu`` suffices. If threads are on different devices, the
+``.sys`` scope must be used. Once a tensormap has been acquired, it can be used by
+threads in the block, after sufficient synchronization, e.g., using
+``__syncthreads()``. The thread that uses the tensormap and the thread that
+performs the fence must be in the same block. The fence **cannot be delegated**
+to a thread in another block or kernel launch. If there are not intermediate
+modifications, the fence does not have be repeated before each
+``cp.async.bulk.tensor`` instruction.
+
+The ``fence`` and subsequent use of the tensor map is shown in the example below.
+
+.. literalinclude:: ./device_tensormap_initialization.cu
+    :language: c++
+    :dedent:
+    :start-after: example-begin use
+    :end-before: example-end use
+
+
+Creating a template tensor map using the driver API
+---------------------------------------------------
+
+The following code creates a minimal tensor map that can be further modified on
+device.
+
+.. literalinclude:: ./device_tensormap_initialization.cu
+    :language: c++
+    :dedent:
+    :start-after: example-begin make-template
+    :end-before: example-end make-template
+
+The ``get_cuTensorMapEncodeTiled`` function is omitted from the code example. It
+can be found in the CUDA Programming Guide or in the full code example linked
+below.
+
+Full code example
+-----------------
+
+The full code example is included below. The `cuda::ptx` instructions in this
+tutorial have become available in CUDA Toolkit 12.5.
```

Comment:
Godbolt does not yet have CUDA 12.5, so there is no link here. I didn't want to link to a broken example.